extension/src/main/kotlin/io/holunda/camunda/bpm/extension/jgiven/SelfUpdatingInstanceSupplier.kt

```diff
@@ -0,0 +1,26 @@
+package io.holunda.camunda.bpm.extension.jgiven
+
+import org.camunda.bpm.engine.RuntimeService
+import org.camunda.bpm.engine.runtime.ProcessInstance
+import java.util.function.Supplier
+
+/**
+ * Implementation holding one process instance id and loading it from runtime service on every request.
+ * This class is intended to be subclassed.
+ */
+open class SelfUpdatingInstanceSupplier(
+  protected val processInstanceId: String,
+  protected val runtimeService: RuntimeService
+) : Supplier<ProcessInstance> {
+
+  /**
+   * Retrieves the process instance.
+   */
+  override fun get(): ProcessInstance {
+    val instances = runtimeService.createProcessInstanceQuery().active().processInstanceId(processInstanceId).list()
+    return when (instances.size) {
+      1 -> instances.first()
```

Comment:
Isn't the size check just a fancy way of `.single() ?: throw ... `?
single() fails when more than 1 instance is found, but that is not possible when we query by id. Its 0 or 1.

Reply:
agree